inject-kotlin/src/main/kotlin/io/micronaut/kotlin/processing/visitor/KotlinMethodElement.kt

```diff
@@ -0,0 +1,379 @@
+        this.internal = internal
+    }
+
+    override fun getOwningType(): ClassElement {
+        return owningType
+    }
+
+    override fun isSynthetic(): Boolean {
+        return if (declaration is KSPropertyGetter || declaration is KSPropertySetter) {
+            return true
+        } else {
+            if (declaration is KSFunctionDeclaration) {
+                return declaration.functionKind != FunctionKind.MEMBER && declaration.functionKind != FunctionKind.STATIC
+            } else {
+                return false
+            }
+        }
+    }
+
+    override fun isFinal(): Boolean {
+        return if (declaration is KSPropertyGetter || declaration is KSPropertySetter) {
+            true
+        } else  {
+            super<AbstractKotlinElement>.isFinal()
+        }
+    }
+
+    override fun getModifiers(): MutableSet<ElementModifier> {
+        return super<AbstractKotlinElement>.getModifiers()
+    }
+
+    override fun getDeclaredTypeVariables(): MutableList<out GenericPlaceholderElement> {
+        val nativeType = kspNode()
+        return if (nativeType is KSDeclaration) {
+            nativeType.typeParameters.map {
+                KotlinGenericPlaceholderElement(it, annotationMetadataFactory, visitorContext)
+            }.toMutableList()
+        } else {
+            super.getDeclaredTypeVariables()
+        }
+    }
+
+    override fun isSuspend(): Boolean {
+        val nativeType = nativeType
+        return if (nativeType is KSModifierListOwner) {
+            nativeType.modifiers.contains(Modifier.SUSPEND)
+        } else {
+            false
+        }
+    }
+
+    override fun getSuspendParameters(): Array<ParameterElement> {
+        val parameters = getParameters()
+        return if (isSuspend) {
+            val continuationParameter = visitorContext.getClassElement("kotlin.coroutines.Continuation")
+                .map {
+                    var rt = genericReturnType
+                    if (rt.isPrimitive && rt.name.equals("void")) {
+                        rt = ClassElement.of(Unit::class.java)
+                    }
+                    val resolvedType = it.withTypeArguments(mapOf("T" to rt))
+                    ParameterElement.of(
+                        resolvedType,
+                        "continuation"
+                    )
+                }.orElse(null)
+            if (continuationParameter != null) {
+
+                ArrayUtils.concat(parameters, continuationParameter)
+            } else {
+                parameters
+            }
+        } else {
+            parameters
+        }
+    }
+
+    override fun overrides(overridden: MethodElement): Boolean {
+        val nativeType = kspNode()
+        val overriddenNativeType = overridden.kspNode()
+        if (nativeType == overriddenNativeType) {
+            return false
+        } else if (nativeType is KSFunctionDeclaration) {
+            return overriddenNativeType == nativeType.findOverridee()
```

Comment:
this only checks the direct parent, it doesnt work if youre overriding an overridden method

Reply:
not sure how to fix this or if it is an issue